safe_landing_planner/include/safe_landing_planner/waypoint_generator.hpp

```diff
@@ -0,0 +1,115 @@
+#pragma once
+
+#include <avoidance/usm.h>
+#include <safe_landing_planner/grid.hpp>
+
+#include <Eigen/Dense>
+
+#include <functional>
+#include <vector>
+
+namespace avoidance {
+
+const std::vector<Eigen::Vector2f> exploration_pattern = {
+    Eigen::Vector2f(1.f, 0.f),  Eigen::Vector2f(1.f, 1.f),
+    Eigen::Vector2f(0.f, 1.f),  Eigen::Vector2f(-1.f, 1.f),
+    Eigen::Vector2f(-1.f, 0.f), Eigen::Vector2f(-1.f, -1.f),
+    Eigen::Vector2f(0.f, -1.f), Eigen::Vector2f(1.f, -1.f)};
+
+enum class SLPState {
+  goTo,
+  loiter,
+  land,
+  altitudeChange,
+};
+std::string toString(SLPState state);  // for logging
+
+class WaypointGenerator : public usm::StateMachine<SLPState> {
+ public:
+  WaypointGenerator();
+  virtual ~WaypointGenerator() = default;
+
+  /**
+  * @brief     computes the setpoints to be sent to the FCU
+  **/
+  void calculateWaypoint();
+
+ protected:
+  // config
+  float yaw_setpoint_ = NAN;
```

Comment:
Since I just  spend an enormous amount of time debugging different conventions for yaw, can we specify (at least in comments here, preferably in the variable name) the frame and the units? (I'm guessing here it's fcu frame and degrees, so maybe `yaw_setpoint_fcu_deg_`?

Reply:
It's in the local frame in radians, it seems we don't have the same complications here that we would in the local_planner because we don't have a histogram.

Reply:
sure, but units might make sense still